src/components/TimePicker/index.js

```diff
@@ -1,185 +1,149 @@
         if (shouldOpenModal) {
-            this.setState({ isOpen: true });
+            setIsOpen(true);
         }
-    }
+    };
 
-    handleClick(event) {
-        const { onClick, readOnly } = this.props;
+    const handleClick = event => {
         if (!readOnly) {
-            this.setState({ isOpen: true });
+            setIsOpen(true);
             onClick(event);
         }
-    }
-
-    handleBlur() {
-        const { onBlur, value } = this.props;
-        onBlur(value);
-    }
-
-    handleFocus() {
-        const { onFocus, value } = this.props;
-        onFocus(value);
-    }
-
-    closeModal() {
-        this.setState({ isOpen: false });
-    }
-
-    /**
-     * Sets focus on the element.
-     * @public
-     */
-    focus() {
-        this.inputRef.current.focus();
-    }
-
-    /**
-     * Sets click on the element.
-     * @public
-     */
-    click() {
-        this.inputRef.current.click();
-    }
-
-    /**
-     * Sets blur on the element.
-     * @public
-     */
-    blur() {
-        this.inputRef.current.blur();
-    }
-
-    render() {
-        const {
-            placeholder,
-            label,
-            required,
-            style,
-            className,
-            labelAlignment,
-            hideLabel,
-            name,
-            bottomHelpText,
-            isCentered,
-            error,
-            readOnly,
-            disabled,
-            tabIndex,
-            id,
-            cancelLabel,
-            okLabel,
-            onChange,
-            hour24,
-        } = this.props;
-        const { isOpen, value } = this.state;
-
-        return (
-            <StyledContainer id={id} className={className} style={style}>
-                <Input
-                    id="time-picker_time-input"
-                    ref={this.inputRef}
-                    label={label}
-                    placeholder={placeholder}
-                    icon={<ClockIcon />}
-                    iconPosition="right"
-                    required={required}
-                    value={this.getTriggerInputValue()}
-                    onKeyDown={this.handleKeyDown}
-                    onClick={this.handleClick}
-                    onFocus={this.handleFocus}
-                    onBlur={this.handleBlur}
-                    labelAlignment={labelAlignment}
-                    hideLabel={hideLabel}
-                    name={name}
-                    bottomHelpText={bottomHelpText}
-                    isCentered={isCentered}
-                    error={error}
-                    readOnly={readOnly}
-                    disabled={disabled}
-                    tabIndex={tabIndex}
+    };
+
+    const handleBlur = () => {
+        onBlur(props.value);
+    };
+
+    const handleFocus = () => {
+        onFocus(props.value);
+    };
+
+    const closeModal = () => {
+        setIsOpen(false);
+    };
+    useEffect(() => {
+        if (prevValue !== props.value) {
+            updateValue();
+        }
+    });
```

Comment:
this should be used to substitute the  `componentDidUpdate` functionality, then you don't need to maintain the prevValue by yourself, instead you can use:
```
useEffect(() => {
    updateValue();
}, [valueProp]);
```
this do the same since it will enter in the useEffect every time the value prop change